src/main/java/com/facebook/openwifirrm/modules/ConfigManager.java

```diff
@@ -364,9 +376,47 @@ public boolean removeConfigListener(String id) {
 		return (configListeners.remove(id) != null);
 	}
 
-	/** Interrupt the main thread, possibly triggering an update immediately. */
-	public void wakeUp() {
-		eventFlag.set(true);
+	/**
+	 * Mark the zone to be updated, then interrupt the main thread to possibly
+	 * trigger an update immediately.
+	 *
+	 * @param zone non-null zone (i.e., venue)
+	 */
+	public void queueZoneAndWakeUp(String zone) {
+		if (zone == null) {
+			logger.debug("Zone to queue must be a non-null String.");
+		}
+		zonesToUpdate.add(zone);
+		wakeUp();
+	}
+
+	/**
+	 * Track all zones to be updated, then interrupt the main thread to possibly
+	 * trigger an update immediately.
+	 */
+	public void queueAllZonesAndWakeUp() {
+		/*
+		 * zonesToUpdate.addAll(...) is not atomic. If it were used, it is
+		 * possible that: only some zones are added before the main thread
+		 * enters its update loop, completes it, and goes to sleep. Then, a zone
+		 * that was initially in zonesToUpdate (but not anymore) may be added
+		 * again (as part of the addAll(...)). Then, an interrupt would trigger
+		 * an update with the latest configs for this zone, but the update would
+		 * not work due to the debounce timer. Therefore, we make it a point to
+		 * copy the state of zonesToUpdate and only add zones that are not
+		 * already in zonesToUpdate at that time.
+		 */
+		Set<String> copy = new HashSet<>(zonesToUpdate);
```

Comment:
Am I misreading this, or does it not actually do anything useful? If we hit the debounce timer then config doesn't get sent (otherwise harmless), but you are just avoiding the logic altogether (and config still doesn't get sent).